reddit.js

```diff
@@ -1,5 +1,52 @@
 var reddit = require("./data.json")
 
+
 // Write your solutions below
 
-console.log(reddit["data"]);
\ No newline at end of file
+/*Part 1 - forEach - console.log all titles*/
+
+reddit.data.children.forEach(function (theTitle){
+	console.log(theTitle.data.title);
+})
+
+/*Part 2 - map - Return an array of permalinks,  to each so that they contain full http:// path to reddit*/
+var permalinks = [];
+
+reddit.data.children.forEach(function (child){
+	permalinks.push(child.data.permalink);
+});
+
+var permalinksHTTP = permalinks.map(function(http){
+	return "http://reddit.com" + http;
+})
+console.log(permalinksHTTP);
+
+/*Part 3 - filter - Filter the posts that contain actual text in the selftext key*/
+var nonEmptySelfText = [];
+var selfText = [];
+
+function empty(text) {
+	return text != "";
+}
+
+reddit.data.children.forEach(function (child){
+	selfText.push(child.data.selftext);
+});
+
+nonEmptySelfText = selfText.filter(empty);
+
+console.log(nonEmptySelfText);
```

Comment:
This way works, but involves a lot of steps. A shortcut would be to simply filter the ```data.children``` array, returning only children that have selftext with a length greater than 0:
```
var filtered = reddit.data.children.filter(function (child) {
	return child.data.selftext.length;
});

console.log(filtered);
```